Firestore/core/src/remote/remote_event.h

```diff
@@ -246,11 +248,11 @@ class TargetState {
 class RemoteEvent {
  public:
   using TargetChangeMap = std::unordered_map<model::TargetId, TargetChange>;
-  using TargetSet = std::unordered_set<model::TargetId>;
+  using TargetMismatchMap = std::map<model::TargetId, local::QueryPurpose>;
```

Comment:
here, original TargetSet used unordered_set
in Web SDK, it used SortedSet:https://github.com/firebase/firebase-js-sdk/blob/eb049456555122e2a7350449b42ac07b8de175f7/packages/firestore/src/remote/remote_event.ts#L49
in android SDK, it used Set: https://github.com/firebase/firebase-android-sdk/blob/687e079d401db1a75d2559038b879726a476ca3e/firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteEvent.java#L30

Should I use a SortedMap here?

Reply:
No. We are moving away from SortedSet/SortedMap and the previous implementation used an unsorted set. Good question though.

Actually, please use `std::unordered_map` instead of `std::map` because `std::unordered_map` has much better performance because it is a "hash map", compared to `std::map` which is a "linked map".